napari/layers/labels/_labels_key_bindings.py

```diff
@@ -74,3 +74,15 @@ def undo(layer):
 def redo(layer):
     """Redo any previously undone actions."""
     layer.redo()
+
+
+@Labels.bind_key('Shift')
+def overwrite(layer):
+    """Disallow overwrite of labeling when pressed"""
+    # on key press
+    layer.overwrite = False
+
+    yield
+
+    # on key release
+    layer.overwrite = True
```

Comment:
```suggestion
    
    current = layer.overwrite
    layer.overwrite = not current
    yield
    # on key release
    layer.overwrite = current
```
This would let the shift key toggle the current setting, rather than "do nothing" if overwrite is already False.  Note: this requires fixing the return type of the `@property`

Reply:
I have been debating with myself whether we should make it a configurable in the panel, similar to contiguous/n-dim, or we change the painting behavior while user keeps the key pressed, if the later then without the key press, we would always default the value to true, so we would never see use of shift as a negate

Reply:
Not everyone will be strictly interacting via the GUI.  Is there any specific reason to _disallow_ someone from programatically setting `layer.overwrite = False` (as you did in the test).  If we allow that to be set programmatically (as it currently is), then at the very least i think that pressing shift shouldn't undo their setting... but I also think using shift to toggle (rather than assert) the behavior is a nice feature.

Reply:
I think `Shift` as a toggle is a nice approach, and someone can set the behavior they want to be unmodified. As a default we should overwrite.

Reply:
We can always revisit this with more advance UX testing, but for now I think it will be fine

Reply:
I made it a switch in the panel in the end, during my testing I found it hard to tell if I am in overwrite mode or not if there is no explicit information. so I made it an explicit toggle